source/common/stats/statsd.cc

```diff
@@ -114,10 +157,12 @@ void TcpStatsdSink::TlsSink::onEvent(uint32_t events) {
 
 void TcpStatsdSink::TlsSink::onTimespanComplete(const std::string& name,
                                                 std::chrono::milliseconds ms) {
-  write(fmt::format("envoy.{}:{}|ms\n", name, ms.count()));
+  // Ultimately it would be nice to perf optimize this path also, but it's not very frequent.
+  Buffer::OwnedImpl buffer(fmt::format("envoy.{}:{}|ms\n", name, ms.count()));
+  write(buffer);
```

Comment:
This actually can't happen in the code. Flushing is basically an inline operation always. I will add asserts.